### PR TITLE
Implement optional vault exporter for prometheus scraping.

### PR DESCRIPTION
**Pull request by dimitrigraf**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: incubator/vault/templates/deployment.yaml [hunk 1]
```diff
@@ -89,6 +89,29 @@ spec:
         resources:
 {{ toYaml .Values.resources | indent 10 }}
       {{- if .Values.affinity }}
+      {{- if .Values.vaultExporter.enabled }}
```
Comment (born4new) [inline incubator/vault/templates/deployment.yaml]: I think it's wrong that there is an `if` statement above, a the exact same indentation...Is it needed?

Comment (dimitrigraf) [reply incubator/vault/templates/deployment.yaml]: Hmm, I'm not sure. It was there before, just above the consul inclusion condition. It looks weird to me, too.

Comment (born4new) [reply incubator/vault/templates/deployment.yaml]: I see...https://github.com/dimitrigraf/charts/blob/master/incubator/vault/templates/deployment.yaml#L122

Comment (dimitrigraf) [reply incubator/vault/templates/deployment.yaml]: Position of vaultExporter block was wrong. I fixed it with the second commit, putting it at the end just before the volume definitions.

**Pull request merged**